File: DesignPatternsExercises/java/PrototypeDesignPattern/Solution/PrototypeFactory.java
public class PrototypeFactory {
    // public __________(6)__________ PrototypeCloneable __________(7)__________(PrototypeCloneable cloneableObject) throws CloneNotSupportedException {
    public static PrototypeCloneable getInstance(PrototypeCloneable cloneableObject) throws CloneNotSupportedException {
        // return cloneableObject.__________(8)__________();
        return cloneableObject.clone();
    }
}

File: DesignPatternsExercises/ruby/prototype_design_pattern/solution/prototype_factory.rb
class PrototypeFactory
  # def __________(3)__________(cloneable_object)
  def self.instance(cloneable_object)
    puts "Cloning #{cloneable_object.class} object..."
    # __________(4)__________.clone
    cloneable_object.clone
  end
end
